proxy/http2/Http2ClientSession.cc

```diff
@@ -326,6 +327,9 @@ Http2ClientSession::main_event_handler(int event, void *edata)
     break;
 
   case VC_EVENT_WRITE_COMPLETE:
+    if (this->get_half_close_flag()) {
```

Comment:
One tiny change needs to be made related to the graceful shutdown. When sending the shutdown notice (stream_id == INT_MAX, error==NO_ERROR), we can't schedule session close. We can only schedule it if `stream_id < INT_MAX`.